spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/AppRegistry.java

```diff
@@ -235,4 +238,12 @@ else if (!StringUtils.hasText(uri.getSchemeSpecificPart())) {
 		}
 		return uri;
 	}
+
+	private void validateApplicationUri(URI uri){
+		if (uri.getScheme().toLowerCase().equals("file")){
+			File file = new File(uri);
+			Assert.isTrue(file.exists() && !file.isDirectory() && file.getName().toLowerCase().endsWith(".jar"),
```

Comment:
Moreover, file *could* be something else than a boot app (could be a python app ultimately, etc)